app/views/topics/show.html.erb

```diff
@@ -23,11 +23,19 @@
             submitted <%= time_ago_in_words(post.created_at) %> ago by <%= post.user.name %> <br>
             <%= post.comments.count %> comments
           </small>
+          <small>
+          <%= render @topic.comments %>
+          </small>
         </div>
       </div>
     <% end %>
+    <div>
+      <% if current_user %>
+        <%= render 'comments/form_topic', comment: Comment.new, topic: @topic %>
```

Comment:
@membrq Its possible to use a single `comments/_form` partial for both topic comments and post comments. Try using a more flexible variable name for the comment owner, e.g. `commentable`:

```diff
# File: `app/views/topics/show.html.erb`
-<%= render 'comments/form', comment: Comment.new, topic: @topic %>
+<%= render 'comments/form', comment: Comment.new, commentable: @topic %>
```

```diff
# File: `app/views/posts/show.html.erb`
-<%= render 'comments/form', comment: Comment.new, post: @post %>
+<%= render 'comments/form', comment: Comment.new, commentable: @post %>
```

Then use `commentable` in the `comments/_form` partial:

```diff
-<%= form_for [post, comment] do |f| %>
+<%= form_for [commentable, comment] do |f| %>
```